### ISSUE TITLE
Volumes attached twice

### ISSUE THREAD

**Issue by nagajagan** [airshipit/treasuremap#4]
```
$ openstack server show b511e47c-7c11-446d-b2e5-61ecc8df7447 -c volumes_attached
+------------------+-------------------------------------------+
| Field            | Value                                     |
+------------------+-------------------------------------------+
| volumes_attached | id='fb9977c6-7f84-4c40-8523-83f20a7b4846' |
|                  | id='1d07f500-93b1-44c5-9dc3-78cda0ca32d7' |
|                  | id='d7302fe1-c535-4600-bc17-9d6197287479' |
|                  | id='d7302fe1-c535-4600-bc17-9d6197287479' |
|                  | id='b4eb50fb-d281-4952-ad0b-c144da765aff' |
|                  | id='768671cf-6beb-4217-9d68-51b7c85b1287' |
|                  | id='b4eb50fb-d281-4952-ad0b-c144da765aff' |
|                  | id='768671cf-6beb-4217-9d68-51b7c85b1287' |
|                  | id='c7bda4ec-ca1c-4a01-9217-c97eb63f227c' |
|                  | id='4aa286bb-0f3f-4c3c-9f04-f5d2ae9634f6' |
|                  | id='96b514c1-1f55-4cae-9161-ba1ac8a4d23b' |
|                  | id='11d7a75c-f666-4554-897e-4bebe48e3b25' |
|                  | id='c7bda4ec-ca1c-4a01-9217-c97eb63f227c' |
|                  | id='4aa286bb-0f3f-4c3c-9f04-f5d2ae9634f6' |
|                  | id='96b514c1-1f55-4cae-9161-ba1ac8a4d23b' |
|                  | id='11d7a75c-f666-4554-897e-4bebe48e3b25' |
|                  | id='102bc4a5-61b7-46cf-b119-17271dfed4ae' |
|                  | id='4ec89665-796d-4d71-9cef-4c3c0be649e3' |
|                  | id='f34dc700-0625-4b96-bbf3-c5aaa9d5a0bd' |
|                  | id='254bd202-6027-4874-9971-df5e56d8e70c' |
|                  | id='55e12f21-e809-4c1e-abb7-3700504c1d44' |
|                  | id='3ec26554-5807-4889-afe0-e2a1af8bd7cd' |
|                  | id='0c3a4bba-53ac-41e8-b4e3-88da7e44d75d' |
|                  | id='61846c8e-23c8-420d-9e26-0d47cddc5457' |
|                  | id='9addf0bb-f7c5-4ac8-8e40-3a19054e14e1' |
+------------------+-------------------------------------------+
```
Please observe there are 7 volumes those are attached twice.   

Please let me know if you need more information.

**Comment by nagajagan** [comment]
Is there any workaround to delete these duplicate attachments?

**Comment by Mihalych** [comment]
If it is not OpenStack general question, can you refer to treasuremap version and when it happened during deployment process? What type of disks do you use? Do you see such issues on different hardware?

**Comment by nagajagan** [comment]
treasure map version 1.3
We have only one lab and we see this issue there.
We are using PowerEdge R740xd  servers. it has TOSHIBA disks.

**Comment by imarijitbose** [comment]
Identified some nova database inconsistencies. Cinder database is correct and consistent. This could be result of some race condition of immediate volume create and attaching them to instances. Manually the issue could not be reproduced. The workaround is provided to delete the rows of size "NULL" from "block_device_mapping" table of nova database.

**Comment by nagajagan** [comment]
Thank you @imarijitbose that worked for me, I don't see duplicate volumes on the server now.

**Comment by imarijitbose** [comment]
Thanks for the confirmation.. Fixes #4

**Comment by nagajagan** [comment]
Thank you for the support.
